### PR TITLE
Simplified introspection of constraints on MySQL.

### PR DESCRIPTION
**Pull request by felixxm**

### CODE REVIEW

**Comment by hannseman** [comment]
Sorry, I missed to run the patch on MariaDB. I can take a look at this in the weekend if you don’t get at it before that.

File: django/db/backends/mysql/introspection.py [hunk 1]
```diff
@@ -211,42 +211,37 @@ def get_constraints(self, cursor, table_name):
...
-            SELECT c.constraint_name, c.constraint_type
-            FROM information_schema.table_constraints AS c
-            WHERE
-                c.table_schema = DATABASE() AND
-                c.table_name = %s
-        """
-        cursor.execute(type_query, [table_name])
-        for constraint, kind in cursor.fetchall():
-            if kind.lower() == "primary key":
+            elif kind == 'PRIMARY KEY':
```
Comment (felixxm) [inline django/db/backends/mysql/introspection.py:239]: Unfortunately unnamed unique and check columns constraints have the same name as a column (see 579909a13feb958f75d1d77e9995cef8f2b9c2a3) so we need to keep `elif` blocks.

File: django/db/backends/mysql/introspection.py [hunk 2]
```diff
@@ -211,42 +211,37 @@ def get_constraints(self, cursor, table_name):
...
-            FROM information_schema.key_column_usage AS kc
+                kc.`referenced_table_name`, kc.`referenced_column_name`,
+                c.`constraint_type`
+            FROM
+                information_schema.key_column_usage AS kc,
+                information_schema.table_constraints AS c
             WHERE
                 kc.table_schema = DATABASE() AND
+                c.table_schema = kc.table_schema AND
+                c.constraint_name = kc.constraint_name AND
```
Comment (hannseman) [inline django/db/backends/mysql/introspection.py:222]: Would it make sense to exclude check constrains from this query and get rid of the `elif` blocks?

```suggestion
                c.table_schema = kc.table_schema AND
                c.constraint_name = kc.constraint_name AND
                c.constraint_type != 'CHECK' AND
```

Comment (felixxm) [reply django/db/backends/mysql/introspection.py:222]: Updated :+1:

**Review by hannseman — APPROVED** [review]
👍

**Pull request merged**